src/inventory_ui.cpp

```diff
@@ -276,24 +277,58 @@ void inventory_selector::print_inv_weight_vol(int weight_carried, int vol_carrie
         int vol_capacity) const
 {
     // Print weight
-    mvwprintw(w_inv, 0, 32, _("Weight (%s): "), weight_units());
-    nc_color weight_color;
+    mvwprintw(w_inv, 0, TERMX - 41, _("Weight: "), weight_units());
+    nc_color weight_color = c_ltgray;
     if (weight_carried > u.weight_capacity()) {
         weight_color = c_red;
-    } else {
-        weight_color = c_ltgray;
     }
     wprintz(w_inv, weight_color, "%6.1f", convert_weight(weight_carried) + 0.05 ); // +0.05 to round up;
-    wprintz(w_inv, c_ltgray, "/%-6.1f", convert_weight(u.weight_capacity()));
+    wprintz(w_inv, c_ltgray, "/%-6.1f %s", convert_weight(u.weight_capacity()), weight_units());
+    wmove(w_inv, 0, TERMX - 15);
+    wprintz(w_inv, c_ltgray, "[............]");
+    print_bar(0, TERMX - 14, weight_carried, g->u.weight_capacity());
 
     // Print volume
-    mvwprintw(w_inv, 0, 61, _("Volume: "));
+    mvwprintw(w_inv, 1, TERMX - 41, _("Volume: "));
+    wmove(w_inv, 1, TERMX - 30);
     if (vol_carried > vol_capacity) {
         wprintz(w_inv, c_red, "%3d", vol_carried);
     } else {
         wprintz(w_inv, c_ltgray, "%3d", vol_carried);
     }
     wprintw(w_inv, "/%-3d", vol_capacity);
+    wmove(w_inv, 1, TERMX - 15);
+    wprintz(w_inv, c_ltgray, "[............]");
+    print_bar(1, TERMX - 14, vol_carried, vol_capacity);
+}
+
+void inventory_selector::print_bar(int y, int x, int carried, int capacity) const {
+    wmove(w_inv, y, x);
+    if(capacity == 0) {
+        wprintz(w_inv, c_red, "------------");
+        return;
+    }
+    
+    int barsize = static_cast<int>((carried * 24.0) / capacity);
+    nc_color color = c_green;
+    if(barsize > 4) { color = c_ltgreen; }
+    if(barsize > 9) { color = c_yellow; }
+    if(barsize > 13) { color = c_ltred; }
+    if(barsize > 18) { color = c_red; }
+    
+    std::stringstream bar;
```

Comment:
You can use ```std::ostringstream``` instead of ```std::stringstream``` since you only use ```<<``` here.
Or even more: you can replace this stream stuff & ```wprintz()``` by using of ```wputch()``` (which puts a single character to the current position set by ```wmove()``` in the beginning of current function.